pkg/reporter/gitlab.go

```diff
@@ -0,0 +1,358 @@
+	Direct  bool          `json:"direct"`
+}
+
+// gitLabLocation represents location information
+type gitLabLocation struct {
+	File       string           `json:"file"`
+	Dependency gitLabDependency `json:"dependency"`
+}
+
+// gitLabIdentifierType represents type of identifier
+// Docs: https://docs.gitlab.com/development/integrations/secure/#identifiers
+type gitLabIdentifierType string
+
+const (
+	gitLabIdentifierTypeCVE  gitLabIdentifierType = "cve"
+	gitLabIdentifierTypeCWE  gitLabIdentifierType = "cwe"
+	gitLabIdentifierTypeGHSA gitLabIdentifierType = "ghsa"
+	// NOT GITLAB BUT WE ARE USING THIS FOR OUR CUSTOM IDENTIFIER
+	gitLabIdentifierTypeMALWARE gitLabIdentifierType = "malware"
+)
+
+// gitLabIdentifier represents identifier information
+type gitLabIdentifier struct {
+	Type  gitLabIdentifierType `json:"type"`
+	Name  string               `json:"name"`
+	Value string               `json:"value"`
+	URL   string               `json:"url"`
+}
+
+// Severity represents severity of a vulnerability or malware
+type Severity string
+
+const (
+	SeverityUnknown  Severity = "Unknown"
+	SeverityCritical Severity = "Critical"
+	SeverityHigh     Severity = "High"
+	SeverityMedium   Severity = "Medium"
+	SeverityLow      Severity = "Low"
+)
+
+// gitLabVulnerability represents a vulnerability in GitLab format
+// Docs: https://docs.gitlab.com/development/integrations/secure/#vulnerabilities
+type gitLabVulnerability struct {
+	ID          string             `json:"id"`
+	Name        string             `json:"name"`
+	Description string             `json:"description"`
+	Severity    Severity           `json:"severity"`
+	Location    gitLabLocation     `json:"location"`
+	Identifiers []gitLabIdentifier `json:"identifiers"`
+}
+
+// gitLabScan represents scan information
+type gitLabScan struct {
+	Scanner   gitLabScanner `json:"scanner"`
+	Analyzer  gitLabScanner `json:"analyzer"` // Reusing GitLabScanner as they have same structure
+	Type      string        `json:"type"`
+	StartTime string        `json:"start_time"`
+	EndTime   string        `json:"end_time"`
+	Status    string        `json:"status"`
+}
+
+// gitLabReport represents the complete GitLab report currently using the 15.2.1 schema
+// and `dependency_scanning` type.
+// but can be extended to support other types and schemas in the future.
+// docs: https://docs.gitlab.com/development/integrations/secure/#report
+type gitLabReport struct {
+	Schema          string                `json:"schema"`
+	Version         string                `json:"version"`
+	Scan            gitLabScan            `json:"scan"`
+	Vulnerabilities []gitLabVulnerability `json:"vulnerabilities"`
+}
+
+type gitLabReporter struct {
+	config          GitLabReporterConfig
+	vulnerabilities []gitLabVulnerability
+	startTime       time.Time
+}
+
+// Ensure gitLabReporter implements Reporter interface
+var _ Reporter = (*gitLabReporter)(nil)
+
+func NewGitLabReporter(config GitLabReporterConfig) (Reporter, error) {
```

Comment:
```suggestion
func NewGitLabReporter(config GitLabReporterConfig) (*gitLabReporter, error) {
```

We can return the internal type here. Since we are using `var _ Reporter = (*gitLabReporter)(nil)` to verify contract. Exposing the internal type helps in testing private functions with tests available in the same package.

Reply:
done